src/devtools/client/inspector/markup/components/rules/RulesPanel.tsx

```diff
@@ -0,0 +1,78 @@
+import { useContext, useDeferredValue, useState } from "react";
+import AutoSizer from "react-virtualized-auto-sizer";
+
+import { getPauseId } from "devtools/client/debugger/src/selectors";
+import { RulesList } from "devtools/client/inspector/markup/components/rules/RulesList";
+import { getSelectedNodeId } from "devtools/client/inspector/markup/selectors/markup";
+import Icon from "replay-next/components/Icon";
+import { ReplayClientContext } from "shared/client/ReplayClientContext";
+import { shallowEqual } from "shared/utils/compare";
+import { useAppSelector } from "ui/setup/hooks";
+import { processedNodeDataCache } from "ui/suspense/nodeCaches";
+import { RuleState, cssRulesCache } from "ui/suspense/styleCaches";
+
+import styles from "./RulesPanel.module.css";
+
+const NO_RULES_AVAILABLE: RuleState[] = [];
+
+export function RulesPanelSuspends() {
+  const replayClient = useContext(ReplayClientContext);
+  const { pauseId, selectedNodeId } = useAppSelector(
+    state => ({
+      pauseId: getPauseId(state),
+      selectedNodeId: getSelectedNodeId(state),
+    }),
+    shallowEqual
+  );
+
+  const deferredPauseId = useDeferredValue(pauseId);
+  const deferredSelectedNodeId = useDeferredValue(selectedNodeId);
+
+  const isPending = pauseId !== deferredPauseId || selectedNodeId !== deferredSelectedNodeId;
+
+  const node = processedNodeDataCache.read(
+    replayClient,
+    deferredPauseId,
+    deferredSelectedNodeId ?? undefined
+  );
+
+  const canHaveRules = !!node?.isElement;
+
+  const cachedStyles = cssRulesCache.read(
+    replayClient,
+    canHaveRules ? deferredPauseId : undefined,
+    canHaveRules ? deferredSelectedNodeId : undefined
+  );
```

Comment:
[This video](https://www.loom.com/share/b1168cda0b674f688a71e2b1a4b3e9f2) explains why I also made this change.